### PR TITLE
Implement calculator

### PR DESCRIPTION
**Pull request by olispe-fdg**

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
```
Comment (olispe-fdg) [inline calculator.js:11]: nit: Can we remove this comment? The error message below should be enough to document this condition

File: calculator.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
```
Comment (olispe-fdg) [inline calculator.js:3]: Could we refactor this to use [ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) instead?

File: calculator.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          default:
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
+        // console.log('current answer: ' + answer);
+
+        i++;
+      });
+
+      return answer;
```
Comment (olispe-fdg) [inline calculator.js:44]: I think we can simplify using [Array.prototype.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) here which would let us remove the `answer` variable.

Worth noting that when we don't specify an initial value for `reduce`, the function will set the previous value to the first element in the array, and the current value will be the second. This means we avoid needing to do `let answer = numbers[0]`

File: calculator.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          case '+':
+            answer = answer + number;
+            break;
+          case '-':
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
```
Comment (olispe-fdg) [inline calculator.js:35]: Should we also handle multiplication here?

File: calculator.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
```
Comment (olispe-fdg) [inline calculator.js:1]: nit: Doesn't look like this variable is being used?

File: calculator.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
+        // console.log('current answer: ' + answer);
```
Comment (olispe-fdg) [inline calculator.js:39]: Can we remove these debug prints please

File: index.js [hunk 7]
```diff
@@ -1,7 +1,20 @@
...
+console.log(`How many numbers do you want to ${operator} ?`);
+const c = parseInt(readline.prompt());
+
+const calculator = calc.create_calculator(operator);
+
+for (var i = 0; i < c; i++) {
+  console.log('Enter a number:');
+  const n = parseInt(readline.prompt());
+  calculator.add_number(n);
+}
```
Comment (olispe-fdg) [inline index.js:18]: I'm not sure this is the best user experience as they're required to know how many numbers they want in advance, and doesn't allow easily correcting mistakes.

I wonder if it would be better to have the user input a comma separate list of numbers instead. What do you think?

File: test/calculator_test.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+const assert = require('assert');
+const chai = require('chai');
+const calc = require('../calculator')
+
+describe('calculator', function () {
+  describe('#calculate()', function () {
```
Comment (olispe-fdg) [inline test/calculator_test.js:6]: nit: I'm not sure these describes are needed?

File: test/calculator_test.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    it('adds one number', function () {
```
Comment (olispe-fdg) [inline test/calculator_test.js:7]: Could we add similar test cases for other supported operators to improve coverage?

This would be a great use of Jest's `describe.each` to reuse these tests for each operator, but unfortunately it doesn't look like mocha/chai has that built in.
```js
[{ operator: "+", expected: 12345 }, ...].forEach(({ operator, expected }) =>
    describe(operator, () => ...)
);
```
⬆️ We can get the same result with this snippet though

We'll want to add some test cases for specific operators too, e.g. division by zero (even if we just assert it produces an invalid value)

File: test/calculator_test.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      calculator.add_number(0);
+      calculator.add_number(-20);
+      calculator.add_number(1486);
+      calculator.add_number(-402);
+      calculator.add_number(4);
+
+      assert.equal(1077, calculator.calculate());
+    });
+
+    it('rejects the calculation when there are no numbers', function () {
```
Comment (olispe-fdg) [inline test/calculator_test.js:38]: Can we have an additional test case for the invalid operator edge case too?

File: index.js [hunk 11]
```diff
@@ -1,7 +1,20 @@
...
+const operator = readline.prompt();
+
+console.log(`How many numbers do you want to ${operator} ?`);
+const c = parseInt(readline.prompt());
+
+const calculator = calc.create_calculator(operator);
+
+for (var i = 0; i < c; i++) {
+  console.log('Enter a number:');
+  const n = parseInt(readline.prompt());
```
Comment (olispe-fdg) [inline index.js:16]: I think we should explicitly validate that the input in an integer, and throw an informative error if not (or re-request the input)
